src/main/java/team/creative/littletiles/client/tool/shaper/LittleToolShaper.java

```diff
@@ -226,10 +226,10 @@ private boolean interact(Level level, Player player, ItemStack stack, BlockHitRe
                         builtShapeConfig);
                 shaper.shapeFinished(level, player, stack, builtSelection, boxes);
                 clearPositions();
-                return true;
+                return false;
             } else if (hit != null) {
                 positions.add(last.copy());
-                return true;
```

Comment:
Not sure why you would return false here? Return true makes sure the action is consumed and all other interactions are cancelled.

Reply:
I turned that to false because the tool repeats too fast in survival mode. Is there any better way to delay this?
(sorry I'm not really good at minecraft modding)

video: https://youtu.be/ZiuElN9GibY